test/e2e/minscale_readiness_test.go

```diff
@@ -138,15 +141,31 @@ func latestRevisionName(t *testing.T, clients *test.Clients, configName string)
 	return config.Status.LatestCreatedRevisionName
 }
 
-func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName string, cond func(int32) bool) error {
-	deployments := clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace)
+func serverlessServicesName(t *testing.T, clients *test.Clients, revisionName string) string {
+	if err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
+		_, err := clients.NetworkingClient.ServerlessServices.Get(revisionName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Error retrieving sks %q: %v", revisionName, err)
+	}
+	sks, err := clients.NetworkingClient.ServerlessServices.Get(revisionName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get sks after it was seen to be live: %v", err)
+	}
+	return sks.Status.PrivateServiceName
+}
+
+func waitForDesiredScale(t *testing.T, clients *test.Clients, privateServiceName string, cond func(int32) bool) error {
+	endpoints := clients.KubeClient.Kube.CoreV1().Endpoints(test.ServingNamespace)
 
 	return wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
-		deployment, err := deployments.Get(deploymentName, metav1.GetOptions{})
+		endpoint, err := endpoints.Get(privateServiceName, metav1.GetOptions{})
 		if err != nil {
 			return false, nil
 		}
-
-		return cond(*deployment.Spec.Replicas), nil
+		return cond(int32(resources.ReadyAddressCount(endpoint))), nil
```

Comment:
Nit: Let's change the `cond` funcs to just take `int64` to avoid unnecessary casts (doesn't matter performance wise, but will pollute the code eventually)